### PR TITLE
fix(api): non obo flow

### PR DESCRIPTION
**Pull request by Orta21**
Ref. metriport/metriport-internal#1785

Ticket: #_[ticket-number]_

### Dependencies

- Upstream: _[this PR points to another PR or depends on its release]_
- Downstream: _[PRs that depend on this one, either point to this or can only be released after this one is released]_

### Description

_[Regular PRs:]_

_[Document your changes, give context for reviewers, add images/videos of UI changes]_

_[Release PRs:]_

- ...
- ...

### Testing

_[Regular PRs:]_

- Local
  - [x] _[Indicate how you tested this, on local or staging]_
  - [x] ...
- Staging
  - [ ] _testing step 1_
  - [ ] _testing step 2_
- Sandbox
  - [ ] _testing step 1_
  - [ ] _testing step 2_
- Production
  - [ ] _testing step 1_
  - [ ] _testing step 2_

_[Release PRs:]_

Check each PR.

### Release Plan

_[This is the release plan for production]_

_[You should execute the exact same steps when releasing to staging to validate it works]_

_[Add and remove items below accordingly]_

- :warning: Points to `master`
- :warning: This contains a DB migration
- [ ] Maintenance window scheduled/created at Checkly
- [ ] Execute this on <env1>, <env2>
  - [ ] _step1_
  - [ ] _step2_
- [ ] Added to [monthly product update](https://www.notion.so/metriport/Customer-Updates-21b4e9d3ad5f4fd68db587a11db28cff?pvs=4) (deprecates a feature that needs to be communicated with customers)
- [ ] Added to [monthly product update](https://www.notion.so/metriport/Customer-Updates-21b4e9d3ad5f4fd68db587a11db28cff?pvs=4) (introduce a feature that would be useful customers)
- [ ] Upstream dependencies are met/released
- [ ] Release NPM packages
- [ ] Fern Definition Updated
- [ ] Release Fern SDKs
- [ ] FFs have been set in Staging, Production, and Sandbox
- [ ] No dependencies between API and Infra that will cause errors during deployment
- [ ] _[action n-1]_
- [ ] _[action n]_
- [ ] Merge this

### CODE REVIEW

File: packages/api/src/external/hie/register-facility.ts [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    oid: facility.oid,
+    contactName: metriportCompanyDetails.name,
+    phone: metriportCompanyDetails.phone,
+    email: metriportEmailForCq,
+    parentOrgOid: cxOid,
+    role: "Connection" as const,
+  });
+}
+
+// WHY DO WE ONLY CREATE BUT WE CREATE OR UPDATE IN CQ?
```
Comment (Orta21) [inline packages/api/src/external/hie/register-facility.ts:275]: Left comment - is this an oversight? Or is there something im missing

Comment (Orta21) [reply packages/api/src/external/hie/register-facility.ts:290]: Does the create also update?

File: packages/api/src/domain/medical/internal-facility.ts
[the remark this answers is not in the document]
Comment (leite08) [reply packages/api/src/domain/medical/internal-facility.ts:1]: And I don't think our domain should be built based off Zod. That's route validation, which is on the interface layer - [reference](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#3c5e5579c5f349d9b29ac1d17dc8472d).

File: packages/api/src/external/hie/register-facility.ts (hunk 1 above)
Comment (leite08) [reply packages/api/src/external/hie/register-facility.ts:297]: It only creates, its mapped as an outstanding item of https://github.com/metriport/metriport-internal/issues/1706

**Pull request closed without merging**